From cbbae89783124302387d2291564f59de22b3f205 Mon Sep 17 00:00:00 2001
From: Sarika Sharma <quic_sarishar@quicinc.com>
Date: Tue, 27 Aug 2024 11:52:18 +0530
Subject: [PATCH] wifi: ath12k: remove warn_on message for peer in mon ring
 process

Currently during monitor ring process, we check for peer using
ath12k_peer_find_by_ast() and  ath12k_peer_find_by_peer_list_index()
while parsing the descriptor info.

When peer gets disconnected due to some reason, it will be removed
from host/firmware entry.
But descriptors are in queue for processing mon ring corresponding
to removed peer as well. So while checking for peer as there will
be no entry of peer for that entry of ids warn_on() prints will
be triggered.

These warn_on() prints will trigger for all remaining descriptors.
This is leading to flooded prints.

Hence, fix by removing warn_on() prints for checking above condition.

Patch-dependency: none

Signed-off-by: Sarika Sharma <quic_sarishar@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp_mon.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

--- a/drivers/net/wireless/ath/ath12k/dp_mon.c
+++ b/drivers/net/wireless/ath/ath12k/dp_mon.c
@@ -3635,11 +3635,8 @@ ath12k_dp_mon_rx_update_user_stats(struc
 		return;
 
 	peer = ath12k_peer_find_by_ast(ar->ab, user_stats->ast_index);
-	if (!peer) {
-		ath12k_warn(ar->ab, "peer with peer id %d can't be found\n",
-			    ppdu_info->peer_id);
+	if (!peer)
 		return;
-	}
 
 	arsta = ath12k_peer_get_link_sta(ar->ab, peer);
 	if (!arsta) {
